content/articles/an-extensive-guide-on-working-with-images-in-django/index.md

```diff
@@ -0,0 +1,424 @@
+
+admin.site.register(Image, imageAdmin)
+```
+As a result, our [Image dashboard](http://127.0.0.1:8000/admin/demo/image/) should resemble this:
+
+![after](/engineering-education/an-extensive-guide-on-working-with-images-in-django/after.jpg)
+
+### Creating Thumbnails
+If you click on images in our [display page](http://127.0.0.1:8000/), they tend to have different default sizes. The big images might even take longer to load due to their size. In case you want to change this behavior and give them a reasonable size, you would have to override the `save` method in our Image model. 
+
+We will create thumbnails of the images by editing their size during the uploading process, and set the maximum value of width and height to 300. Let's update our `models.py` into:
+```py
+from django.db import models
+from django.utils.safestring import mark_safe 
+from PIL import Image as Im                  # new
+
+# Create your models here.
+
+class Image(models.Model):
+    title = models.CharField(max_length=20)
+    photo = models.ImageField(upload_to='pics')
+
+    def image_tag(self):                     
+        return mark_safe('<img src="/../../media/%s" width="150" height="150" />' % (self.photo))
+
+    def save(self):                        # new
+        super().save()
+
+        img = Im.open(self.photo.path)
+
+        # resize it
+        if img.height > 300 or img.width > 300:
+            output_size = (300,300)
+            img.thumbnail(output_size)
+            img.save(self.photo.path)
+
+```
+
+### Adding a background image in a Django template
+We shall go on and add a background image to our upload template. To do so we shall create a directory named `static` in our app level directory `demo`, and add the background image you intend to use inside it. 
+
+Our image in this case is called `bj.jpg`. In `upload.html`, We shall load the statics at the top, then add the background image in the body tag. Update your `upload.html` to:
+```HTML
+{% load static %}
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Image Upload</title>
+</head>
+<body  style="background-image: url('{% static 'bg.jpg' %}');">
+    <form method="POST" enctype="multipart/form-data">
+        {% csrf_token %}
+        {{ form.as_p }}
+        <button type="submit"> Upload Image </button>
+
+    </form>
+     
+</body>
+</html>
+```
+After that, we shall change the `STATIC_URL` in `settings.py` from 
+```py 
+STATIC_URL = 'static/'
+```
+To 
+```py
+STATIC_URL = '/static/'
+```
+This enables Django to find the static files(in this case the background image) within our `static` folder.
+
+### Conclusion
+We have looked at various solutions regarding working with images in Django. I Now hope you will be able to implement these solutions into your Django project as well.
```

Comment:
`at various solutions regarding working with images in Django`

I think your article primarily deals with building a Django application that handles images. But, doesn't discuss various solutions to work with images.

Reply:
Add GitHub repo containing source code